diag_manager/fms_doubly_linked_list.F90

```diff
@@ -0,0 +1,275 @@
+!> \author Miguel Zuniga
+!> \email miguel.zuniga@noaa.gov
+!! \brief A module for a double linked-list.
+!!
+!! \description A double linked list module that is rougly a fortran translation of the C++
+!! boubly linked list from the book  ``Data Structures And Algorithm Analysis in C++", 3rd Edition,
+!!  by Mark Allen Weiss
+module fms_doubly_linked_list_mod
+  implicit none
+
+  !> Linked list node
+  type, private :: node
+    private
+    class(*), pointer :: data => null()
+    type(node), pointer :: next => null()
+    type(node), pointer :: prev => null()
+  end type node
+
+  !> Linked list literator
+  type, public :: literator
+    type(node), pointer :: current
+    type(node), pointer :: end
+  contains
+    procedure :: has_data => literator_has_data
+    procedure :: next => literator_next
+    procedure :: get => literator_data
+  end type literator
+
+  type, public :: linked_list
+    !! Note we are overriding the default constructor with an
+    ! interface of the same name
+    type(node), pointer :: head
+    type(node), pointer :: tail
+    integer :: theSize
```

Comment:
Variables need to be documented in doxygen style

Reply:
Done.